docs/vector-library.md

```diff
@@ -0,0 +1,119 @@
+# Vector library
+
+## Summary
+
+Implement a standard library that provides functionality for the vector type.
+
+## Motivation
+
+Currently, vectors are a primitive type implemented internally. All of the heavy work to implement vectors is done, but there is no runtime-agnostic way to use vectors in Luau code. Individual Luau runtimes, such as [Lune](https://github.com/lune-org/lune) or Roblox, must provide their own distinct libraries to work with native vectors which may be inconsistent. In cross-runtime code, this results in performance drawbacks & difficulty utilizing native vectors.
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions and constants.
+
+`vector.create(x: number, y: number, z: number): vector`
+
+Creates a new vector with the given components. When in 4-wide mode takes a fourth `w` argument. None of the arguments are optional.
```

Comment:
Constructor is something that already has a defined behavior in both compiler and runtime.
This change will not be backwards compatible for 4-wide vector mode.

For 4-wide vectors I propose a different description:

```suggestion
`vector.create(x: number, y: number, z: number): vector`

Creates a new vector with the given components. In 4-wide mode, the forth `w` component value is set to 0.0.

`vector.create(x: number, y: number, z: number, w: number): vector`

Additional constructor available in 4-wide mode. Creates a new vector with the given components.
```

I think this matches the compiler behavior where `vector.create(1, 2, 3, nil)` will not be recognised as a constant and the fast-call behavior where again, `vector.create(1, 2, 3, nil)` will exit the fast-call into a library call fallback. 
`vector.create(1, 2, 3)` is allowed in constant folding and fast-calls in 4-wide mode today.

While we do define the fallback library function in Conformance.test.cpp to treat explicit `nil` as 0.0, we can change the method to use the new library one, since it's just a unit test.